pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go

```diff
@@ -415,6 +419,22 @@ func (a *activeRangeFeed) onRangeEvent(
 	a.RangeID = rangeID
 }
 
+// onConnect is a callback invoked when attempt to connect to the specified
+// destination is made.
+func (a *activeRangeFeed) onConnect(
+	dest rpc.RestrictedInternalClient, metrics *DistSenderRangeFeedMetrics,
+) {
+	if a.onDisconnect != nil {
+		a.onDisconnect()
+	}
```

Comment:
Why are we calling `onDisconnect()` during `onConnect()`?

Reply:
Perhaps I need better name.. Basically, if onConnect() failed and we are trying to connect to another replica,
we need to "disconnect" -- i.e. update metrics to decrement them.

Reply:
Isn't that an artifact of either calling `onConnect()` prematurely (before we know whether the call succeded), or not calling `onDisconnect()` when we lose a connection?

Reply:
Ptal.  Dropped onDisconnnect.  Just a simple bool now.